huaweicloud/services/ddm/resource_huaweicloud_ddm_account.go

```diff
@@ -0,0 +1,518 @@
+	}
+	bodyParams := map[string]interface{}{
+		"user": params,
+	}
+	return bodyParams
+}
+
+func buildUpdateAccountSchemasChildBody(d *schema.ResourceData) []map[string]interface{} {
+	rawParams := d.Get("schemas").([]interface{})
+	if len(rawParams) == 0 {
+		return nil
+	}
+	params := make([]map[string]interface{}, 0)
+	for _, param := range rawParams {
+		perm := make(map[string]interface{})
+		perm["name"] = utils.PathSearch("name", param, nil)
+		params = append(params, perm)
+	}
+	return params
+}
+
+func buildUpdateAccountPasswordBodyParams(d *schema.ResourceData) map[string]interface{} {
+	bodyParams := map[string]interface{}{
+		"password": utils.ValueIngoreEmpty(d.Get("password")),
+	}
+	return bodyParams
+}
+
+func resourceDdmAccountRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	var mErr *multierror.Error
+
+	// getAccount: Query DDM account
+	var (
+		getAccountHttpUrl = "v1/{project_id}/instances/{instance_id}/users"
+		getAccountProduct = "ddm"
+	)
+	getAccountClient, err := cfg.NewServiceClient(getAccountProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating DDM Client: %s", err)
+	}
+
+	parts := strings.SplitN(d.Id(), "/", 2)
+	if len(parts) != 2 {
+		return diag.Errorf("invalid id format, must be <instance_id>/<account_name>")
+	}
+	instanceID := parts[0]
+	accountName := parts[1]
+	getAccountPath := getAccountClient.Endpoint + getAccountHttpUrl
+	getAccountPath = strings.ReplaceAll(getAccountPath, "{project_id}", getAccountClient.ProjectID)
+	getAccountPath = strings.ReplaceAll(getAccountPath, "{instance_id}", fmt.Sprintf("%v", instanceID))
+
+	getAccountResp, err := pagination.ListAllItems(
+		getAccountClient,
+		"offset",
+		getAccountPath,
+		&pagination.QueryOpts{MarkerField: ""})
+
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "error retrieving DdmAccount")
+	}
+
+	getAccountRespJson, err := json.Marshal(getAccountResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	var getAccountRespBody interface{}
+	err = json.Unmarshal(getAccountRespJson, &getAccountRespBody)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	accounts := utils.PathSearch("users", getAccountRespBody, nil)
+	if accounts == nil {
+		return diag.Errorf("error getting DdmAccount, users is not found in API response %s", err)
+	}
+	for _, account := range accounts.([]interface{}) {
+		name := utils.PathSearch("name", account, nil)
+		if accountName != name {
+			continue
+		}
+		mErr = multierror.Append(
+			mErr,
+			d.Set("region", region),
+			d.Set("name", name),
+			d.Set("status", utils.PathSearch("status", account, nil)),
+			d.Set("permissions", utils.PathSearch("base_authority", account, nil)),
+			d.Set("description", utils.PathSearch("description", account, nil)),
+			d.Set("schemas", flattenGetAccountResponseBodyDatabase(account)),
+		)
+		break
+	}
+
+	return diag.FromErr(mErr.ErrorOrNil())
```

Comment:
if the user is not found, we should remove the resource in state

Reply:
done